src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -84,7 +84,7 @@ private boolean validString(Node node, String key) {
     }
 
     private boolean validPath(Node node, String key) {
-        return validString(node, key) && Files.exists(Paths.get(key));
+        return validString(node, key) && Files.exists(Paths.get(Coerce.toString(node)));
```

Comment:
node can be null

Reply:
validString takes care of that, if thats false then second one will not get evaluated.

Reply:
No, it won't. Paths.get will throw a InvalidPathException